pkg/controllers/raycluster_controller.go

```diff
@@ -130,29 +138,56 @@ func (r *RayClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
-	_, err := r.routeClient.Routes(cluster.Namespace).Apply(ctx, desiredClusterRoute(&cluster), metav1.ApplyOptions{FieldManager: controllerName, Force: true})
-	if err != nil {
-		logger.Error(err, "Failed to update OAuth Route")
-	}
+	if cluster.Status.State != "suspended" && r.isRayDashboardOAuthEnabled() && isOpenShift {
+		logger.Info("Creating OAuth Objects")
+		_, err := r.routeClient.Routes(cluster.Namespace).Apply(ctx, desiredClusterRoute(&cluster), metav1.ApplyOptions{FieldManager: controllerName, Force: true})
+		if err != nil {
+			logger.Error(err, "Failed to update OAuth Route")
```

Comment:
I _think_ we should be returning the error with all of these failures. Similar to what Edson mentioned/proposed in https://github.com/project-codeflare/codeflare-operator/pull/498/files.

`return ctrl.Result{RequeueAfter: requeueTime}, err`

Reply:
That makes sense to me, this way we can fail-fast and re-queue. Change has been made. - Thanks Anish!